### PR TITLE
Ensure test suite and file rotation work on Windows

### PR DESCRIPTION
**Pull request by lukebakken**
Follow-up to #493

### CODE REVIEW

**Comment by mrallen1** [comment]
Thanks for taking this on

**Comment by lukebakken** [comment]
Hi @mrallen1!

I have confirmed today that these changes do address the file rotation issues in RabbitMQ on `win32`. I'm continuing to test but I figure getting a review started would be nice.

There is a breaking API change in `lager_rotator_behaviour` in that a file's creation time is passed as an argument. I suppose we could get clever in working around that breaking change if you'd like, or use this as an opportunity to release Lager `4.0.0`.

What do you think?

**Comment by mrallen1** [comment]
Looks like solid work but tbh I'm not a huge fan of so many whitespace changes - all the noise makes it hard to trace what "real" changes have been made. As far as breaking the `lager_rotator_behaviour` I think we can tag a new release as 3.8.0 with appropriate warnings in the README.

It would be great to get a blurb in the README as part of this PR.

**Comment by lukebakken** [comment]
@mrallen1 there was one file formatted with 2-space indentation. Can you give the "ignore whitespace" review link a try before I revert that? 

[`https://github.com/erlang-lager/lager/pull/509/files?w=1`](https://github.com/erlang-lager/lager/pull/509/files?w=1)

I'll add a README updated today. Still testing!

File: test/pr_stacktrace_test.erl [hunk 1]
```diff
@@ -22,56 +22,42 @@ bad_arg() ->
...
         ?EXCEPTION(Class, Reason, Stacktrace) ->
             lager:pr_stacktrace(?GET_STACK(Stacktrace), {Class, Reason})
     end,
-ExpectedPart = "
-    pr_stacktrace_test:pr_stacktrace_throw_test/0 line 26
-    pr_stacktrace_test:make_throw/0 line 16
-throw:{test,exception}",
-    ?assertNotEqual(0, string:str(Result, ExpectedPart)).
-
+    Want = "pr_stacktrace_test:pr_stacktrace_throw_test/0 line 26\n    pr_stacktrace_test:make_throw/0 line 16\nthrow:{test,exception}",
```
Comment (lukebakken) [inline test/pr_stacktrace_test.erl]: When I clone this repo on `win32`, my git settings check out files with `\r\n` endings, which makes this test fail due to embedded `\r` characters in the long string. This is why these expected strings are on a single line now.

**Comment by lukebakken** [comment]
Well, it looks like even the `master` branch has a [heisen-bug in the test suite](https://travis-ci.org/lukebakken/lager/builds/569583114), but only on Travis CI. Everything runs fine on my OS X laptop, Linux and Windows workstations. I'll continue to investigate.

**Comment by michaelklishin** [comment]
Yay, Travis is green 💚

**Comment by mrallen1** [comment]
Thanks again for getting win32 on a good level, Luke. Your contributions are much appreciated!

**Comment by mrallen1** [comment]
Incidentally, I tagged and pushed 3.8.0 to hex just now.

**Comment by lukebakken** [comment]
Thanks @mrallen1 👍

**Comment by lukebakken** [comment]
@mrallen1 if you get a chance, could you enable the lager build on AppVeyor?

https://ci.appveyor.com/project/erlang-lager

**Comment by mrallen1** [comment]
It says "not found" for some reason

**Comment by mrallen1** [comment]
OK, seems to be going now.

**Pull request merged**